### PR TITLE
[New Rule] Interactive Exec Command Launched Against A Running Container

### PR DESCRIPTION
**Pull request by imays11**
## Issues

https://github.com/elastic/detection-rules/issues/2785
### Description

This rule detects interactive 'exec' events launched against a container using the 'exec' command. Using the 'exec'
command in a pod allows a user to establish a temporary shell session and execute any process/command inside the container.
This rule specifically targets higher-risk interactive commands that allow real-time interaction with a container's shell.
A malicious actor could use this level of access to further compromise the container environment or attempt a container breakout.

### Query
```
process where container.id: "*" and event.type== "start" and 
/* use of kubectl exec to enter a container */
process.entry_leader.entry_meta.type: "container" and 
/* process is the inital process run in a container */
process.entry_leader.same_as_process== true and
/* interactive process */
process.interactive== true
```

<details>
<summary/> Screenshot of Query</summary>

image

</details>

### CODE REVIEW

**Comment by imays11** [comment]
@mitodrummer would you mind giving your thoughts on this one? I use a few fields that I'd like to double check that my assumptions are correct and that I'd be capturing what I aim to here. Interactive container shells spawned via `kubectl exec` into a pod.

**Comment by mitodrummer** [comment]
> @mitodrummer would you mind giving your thoughts on this one? I use a few fields that I'd like to double check that my assumptions are correct and that I'd be capturing what I aim to here. Interactive container shells spawned via `kubectl exec` into a pod.

I would drop **process.session_leader.same_as_process**. it's usually the same data (pid) as entry_leader, so wouldn't add anything in most cases. Though if you wanted to detect when an entry session creates a sub session as well, you could use same_as_process from session_leader and drop the entry_leader one. Other than that looks good!

**Comment by imays11** [comment]
> > @mitodrummer would you mind giving your thoughts on this one? I use a few fields that I'd like to double check that my assumptions are correct and that I'd be capturing what I aim to here. Interactive container shells spawned via `kubectl exec` into a pod.
> 
> I would drop **process.session_leader.same_as_process**. it's usually the same data (pid) as entry_leader, so wouldn't add anything in most cases. Though if you wanted to detect when an entry session creates a sub session as well, you could use same_as_process from session_leader and drop the entry_leader one. Other than that looks good!

Thank you! I dropped the session_leader for this one.

File: rules/integrations/cloud_defend/execution_interactive_exec_to_container.toml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+type = "eql"
+
+query = '''
+process where container.id: "*" and event.type== "start" and 
+/* use of kubectl exec to enter a container */
+process.entry_leader.entry_meta.type: "container" and 
+/* process is the inital process run in a container */
+process.entry_leader.same_as_process== true and
+/* interactive process */
+process.interactive== true
```
Comment (Samirbous) [inline rules/integrations/cloud_defend/execution_interactive_exec_to_container.toml:51]:
```suggestion
process where container.id : "*" and event.type== "start" and 

/* use of kubectl exec to enter a container */
process.entry_leader.entry_meta.type : "container" and 

/* process is the inital process run in a container */
process.entry_leader.same_as_process== true and

/* interactive process */
process.interactive == true
```

**Review by Samirbous — APPROVED** [review]

File: rules/integrations/cloud_defend/execution_interactive_exec_to_container.toml [hunk 2]
```diff
@@ -0,0 +1,76 @@
+[metadata]
+creation_date = "2023/05/12"
+integration = ["cloud_defend"]
+maturity = "production"
+min_stack_comments = "New Integration: Cloud Defend"
+min_stack_version = "8.8.0"
+updated_date = "2023/05/12"
```
Comment (imays11) [inline rules/integrations/cloud_defend/execution_interactive_exec_to_container.toml:7]:
```suggestion
updated_date = "2023/05/16"
```

**Pull request merged**